CRM/Contact/BAO/Query.php

```diff
@@ -3293,6 +3293,8 @@ public function tag(&$values) {
     $tagTree = CRM_Core_BAO_Tag::getChildTags();
     foreach ((array) $value as $tagID) {
       if (!empty($tagTree[$tagID])) {
+        // make sure value is an array here (see CORE-2502)
+        $value = is_array($value) ? $value : [$value];
         $value = array_unique(array_merge($value, $tagTree[$tagID]));
```

Comment:
Even cleaner would be

```
$value = array_unique(array_merge((array) $value, $tagTree[$tagID]));
```

Reply:
Yes, that does look neater. Should I change it then?

Reply:
Might as well!

Reply:
Have used your version, and also managed to squash the commits this time, thanks @eileenmcnaughton